File: python/Test_MSE.py
# Copyright 2023 by Xiaoru Liu, Trinity College Dublin. All rights reserved.
#
# This file is the unit test for MSE.
# ==================================================
"Create a function to test MSE"
import unittest
import numpy as np
from Quality_Inspection import MSE_calculation


class Test_MSE(unittest.TestCase):

    def testMSE(self):

        x1 = np.array([[1, 2], [3, 4]])
        y1 = np.array([[4, 3], [2, 1]])
        # expected_output = 5.0
        # output = MSE_calculation(x1, y1)
        # self.assertAlmostEqual(output, expected_output)
        self.assertAlmostEqual(MSE_calculation(x1, y1), 5, places=4)
        x1 = np.array([[12, 11], [10, 10]])
        y1 = np.array([[10, 10], [10, 10]])

        self.assertAlmostEqual(MSE_calculation(x1, y1), 1.25, places=4)

if __name__ == '__main__':
    unittest.main()

File: matlab/Test_MSE.m
classdef Test_MSE < matlab.unittest.TestCase
    methods (Test)
        function testOutputMatrix(test)
            ref_mitrix = rand(5, 5);
            result_mitrix = rand(5, 5);
            expOut = sum((ref_mitrix(:)  -result_mitrix(:)) .^ 2) ...
                / numel(ref_mitrix);
            output = MSE_calculation(ref_mitrix, result_mitrix); 
            maxDiff = 10 ^ (-6);
            diff = abs(output - expOut);
            assertLessThanOrEqual(test, diff, maxDiff);
        end
    end
end
